sdk/core-rest/core-client/src/sendRequest.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  createHttpHeaders,
+  createPipelineRequest,
+  HttpMethods,
+  Pipeline,
+  RawHttpHeaders,
+} from "@azure/core-rest-pipeline";
+import { getCachedDefaultHttpsClient } from "./clientHelpers";
+import { RequestParameters } from "./pathClientTypes";
+import { HttpResponse } from "./common";
+
+/**
+ * Helper function to send request used by the client
+ * @param method method to use to send the request
+ * @param url url to send the request to
+ * @param pipeline pipeline with the policies to run when sending the request
+ * @param options request options
+ * @returns returns and HttpResponse
+ */
+export async function sendRequest(
+  method: HttpMethods,
+  url: string,
+  pipeline: Pipeline,
+  options: RequestParameters = {}
+): Promise<HttpResponse> {
+  const httpClient = getCachedDefaultHttpsClient();
+
+  const headers = createHttpHeaders({
+    accept: "application/json",
+    "content-type": options.contentType || getContentType(options.body),
+    ...(options.headers ? options.headers : {}),
+  });
+
+  const body = JSON.stringify(options.body);
+
+  const request = createPipelineRequest({
+    url: url.toString(),
```

Comment:
why the `toString()`? `url` is already typed as `string`

Reply:
good catch! I think I was doing some URL manipulation here before and forgot to update